sound/soc/intel/boards/Kconfig

```diff
@@ -286,10 +286,10 @@ endif ## SND_SOC_INTEL_SKL
 
 config SND_SOC_INTEL_DA7219_MAX98357A_GENERIC
 	tristate
+	depends on SND_HDA_CODEC_HDMI
```

Comment:
we should only add a 'depends on' on user-visible options.
Otherwise, we have a risk that this is selected but depends on something else, so the depends is ignored.

That should be fine to remove this one since you filter in the top-levels

Reply:
@plbossart Ack. I moved this to SND_SOC_INTEL_BXT_DA7219_MAX98357A_MACH which is user-visible.